### ISSUE TITLE
Adding EVPN EBGP Unnumbered support for SR Linux

### ISSUE THREAD

**Issue by hellt** [ipspace/netlab#1299]
This issue is to track the progress of enabling the `EVPN EBGP Unnumbered` use case support covered by the [`./tests/integration/evpn/15-vxlan-ebgp-unnumbered.yml`](https://github.com/ipspace/netlab/blob/7288143b256a98efabc90c4e765eeafa961ce925/tests/integration/evpn/15-vxlan-ebgp-unnumbered.yml)

The diff that is required on leaf1/spine is:

```
insert / network-instance default protocols bgp afi-safi evpn admin-state enable
insert / network-instance default protocols bgp afi-safi evpn ipv4-unicast advertise-ipv6-next-hops true
insert / network-instance default protocols bgp afi-safi evpn ipv4-unicast receive-ipv6-next-hops true
insert / network-instance default protocols bgp afi-safi ipv4-unicast export-policy default_bgp_export
insert / network-instance default protocols bgp afi-safi ipv4-unicast import-policy accept_all
insert / network-instance default protocols bgp afi-safi ipv4-unicast ipv4-unicast advertise-ipv6-next-hops true
insert / network-instance default protocols bgp afi-safi ipv4-unicast ipv4-unicast receive-ipv6-next-hops true
delete / network-instance default protocols bgp group ebgp export-policy
delete / network-instance default protocols bgp group ebgp afi-safi evpn
delete / network-instance default protocols bgp group ebgp afi-safi evpn admin-state
delete / network-instance default protocols bgp group intf-ethernet-1/1 export-policy
delete / network-instance default protocols bgp group intf-ethernet-1/1 import-policy
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi evpn
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi evpn admin-state
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi ipv4-unicast
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi ipv4-unicast admin-state
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi ipv4-unicast ipv4-unicast
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi ipv4-unicast ipv4-unicast advertise-ipv6-next-hops
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi ipv4-unicast ipv4-unicast receive-ipv6-next-hops
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi ipv6-unicast
delete / network-instance default protocols bgp group intf-ethernet-1/1 afi-safi ipv6-unicast admin-state
delete / network-instance default protocols bgp group intf-ethernet-1/2 export-policy
delete / network-instance default protocols bgp group intf-ethernet-1/2 import-policy
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi evpn
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi evpn admin-state
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi ipv4-unicast
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi ipv4-unicast admin-state
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi ipv4-unicast ipv4-unicast
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi ipv4-unicast ipv4-unicast advertise-ipv6-next-hops
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi ipv4-unicast ipv4-unicast receive-ipv6-next-hops
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi ipv6-unicast
delete / network-instance default protocols bgp group intf-ethernet-1/2 afi-safi ipv6-unicast admin-state
```

The root cause for this diff is:

1. some BGP policies that are designed for IPV4 family, were applied to the BGP global level and thus affecting the EVPN routes as well
2. receive-ipv6-nh and send ipv6 nh for evpn family was not enabled

When the desired state is known, the 2nd step is to understand how to change the templates to enable the proper design.

**Comment by ipspace** [comment]
> 1. some BGP policies that are designed for IPV4 family, were applied to the BGP global level and thus affecting the EVPN routes as well
> 2. receive-ipv6-nh and send ipv6 nh for evpn family was not enabled
> 
> When the desired state is known, the 2nd step is to understand how to change the templates to enable the proper design.

**Comment by hellt** [comment]
> So you're saying we should be moving the **import-policy** and **export-policy** to AF level? That shouldn't be a problem.
> 
> >

The underpinning issue was that a single policy set was applied to all address families. This is the **existing** way how a policy was defined for a given neighbor:

```
/ network-instance default protocols bgp group intf-ethernet-1/2 import-policy
```

The said policy, therefore, was used for all address families; some of the statements designed for v4 afi were messing with evpn af.

The potential fix (potential because it was confirmed in a lab with manual changes, but not with the template change) is to do the following two steps:

1. apply existing policies only to v4 afi, by changing this (both import and export policies)
```
/ network-instance default protocols bgp group intf-ethernet-1/2 import-policy
```
to this
```
/ network-instance default protocols bgp group intf-ethernet-1/2 afi-safi ipv4-unicast import-policy
```

2. add 
```
/ network-instance default protocols bgp afi-safi evpn ipv4-unicast advertise-ipv6-next-hops true
/ network-instance default protocols bgp afi-safi evpn ipv4-unicast receive-ipv6-next-hops true
```

This was tested with the default srl version that netlab currently uses

**Comment by ipspace** [comment]
The only thing that was needed to get the EVPN tests to work with the 24.7.1 image was to move the routing policies to AF level (and a bit of a cleanup in the EVPN template) (see 8c6e2ed22fac80e42a8c5e68513f9ac00b2e4f1e)

I'm guessing the same should be true for 24.3.

Now that I'm almost done with the changes needed to support 24.7 (see #1298 for details), which image would you like to see used in the next netlab release?

**Comment by hellt** [comment]
> Now that I'm almost done with the changes needed to support 24.7 (see #1298 for details), which image would you like to see used in the next netlab release?

ghcr.io/nokia/srlinux:24.7.1 should be fine

**Comment by ipspace** [comment]
@hellt #1300 is ready in case you want to review it (can't add you as a reviewer)

**Comment by ipspace** [comment]
Implemented in #1300
